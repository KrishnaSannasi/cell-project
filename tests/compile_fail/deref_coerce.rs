use cell_project::cell_project as project;
use std::cell::Cell;

pub struct Foo {
    bar: Bar,
    quax: i32,
}

pub struct Bar {
    foo: i32,
}

impl std::ops::Deref for Foo {
    type Target = Bar;

    fn deref(&self) -> &Self::Target {
        &self.bar
    }
}

fn check_deref_coercion(foo: &Foo) -> &i32 {
    &foo.foo
}

fn project(cell: &Cell<Foo>) -> &Cell<i32> {
    project!(Foo, cell.foo)
}

fn main() {}
