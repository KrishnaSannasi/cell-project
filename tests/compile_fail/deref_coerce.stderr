error[E0026]: struct `Foo` does not have a field named `foo`
  --> $DIR/deref_coerce.rs:26:24
   |
26 |     project!(Foo, cell.foo)
   |                        ^^^ struct `Foo` does not have this field
